### PR TITLE
Update petsc4py installation instructions

### PR DESCRIPTION
**Pull request by A-Gray-94**
## Purpose
Update the latest and stable PETSc/petsc4py versions in the third party packages table and changed the petsc4py installation instructions for newer versions of PETSc which include the petsc4py source

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation update
- [ ] Maintenance update
- [ ] Other (please describe)

## Checklist
_Put an `x` in the boxes that apply._

- [ ] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

**Review by nwu63 — CHANGES_REQUESTED** [review]
Can you also update the instructions to use `pip` instead of `python setup.py`, both for single and multi-arch installs (with the colon)?

File: installInstructions/install3rdPartyPackages.rst [hunk 1]
```diff
@@ -445,38 +445,47 @@ It is installed with:
...
 
-From the ``petsc4py`` directory do a user-space install:
+From 3.14 onwards, petsc4py is included in the PETSc source code, in which case you can skip the above step and simply go straight to the petsc4py source directory:
 
 .. prompt:: bash
 
-   python setup.py install --user
+   cd $PETSC_DIR/src/binding/petsc4py
+
+Then do a user-space install:
```
Comment (nwu63) [inline installInstructions/install3rdPartyPackages.rst:467]: Not really a user-space install, I think we can just get rid of the user/system/environment stuff and assume users know what they're doing.

File: installInstructions/install3rdPartyPackages.rst [hunk 2]
```diff
@@ -445,38 +445,47 @@ It is installed with:
...
+
+.. prompt:: bash
+
+   pip install .
 
 This will install the package to the ``.local`` directory in your home directory which is suitable for both desktop and cluster accounts.
```
Comment (nwu63) [inline installInstructions/install3rdPartyPackages.rst:473]: Can remove this line

File: installInstructions/install3rdPartyPackages.rst [hunk 3]
```diff
@@ -445,38 +445,47 @@ It is installed with:
...
 You may seen an error warning related to ``python-mpi``, but this should not be a problem.
```
Comment (nwu63) [inline installInstructions/install3rdPartyPackages.rst:474]: And this is probably outdated too I think

File: installInstructions/install3rdPartyPackages.rst (hunk 1 above)
Comment (A-Gray-94) [reply installInstructions/install3rdPartyPackages.rst]: > assume users know what they're doing.

Never a good idea

Comment (nwu63) [reply installInstructions/install3rdPartyPackages.rst]: Yeah I know. For those in the lab we have a pretty detailed page on setting up python, but for external users it's not really our job to explain how python works.

**Review by nwu63 — APPROVED** [review]

**Pull request merged**